cypher-shell/src/main/java/org/neo4j/shell/commands/Param.java

```diff
@@ -56,13 +58,26 @@ public String getHelp() {
 
     @Override
     public void execute(@Nonnull final String argString) throws CommandException {
-        Matcher m = argPattern.matcher(argString);
+        Matcher alphanumericMatcher = argPattern.matcher(argString);
 
-        if (!m.matches()) {
-            throw new CommandException(AnsiFormattedText.from("Incorrect number of arguments.\nusage: ")
-                    .bold().append(COMMAND_NAME).boldOff().append(" ").append(getUsage()));
+        if (!alphanumericMatcher.matches()) {
```

Comment:
I agree with this being unreadable, I refactored the logic to be better expressed. I have however not implemented in the way that you suggested. Do let me know your thoughts.